src/tools/compiletest/src/header.rs

```diff
@@ -877,7 +879,13 @@ pub fn make_test_description<R: Read>(
             return;
         }
         ignore = match config.parse_cfg_name_directive(ln, "ignore") {
-            ParsedNameDirective::Match => true,
+            ParsedNameDirective::Match => {
+                #[cfg(not(bootstrap))]
+                {
+                    ignore_message = config.parse_name_value_directive(ln, "ignore");
```

Comment:
I don't think this makes sense -- the text after ignore doesn't necessarily follow the name/value directive syntax (indeed, often will not) and even if it does may not mean what we "want" it to. We can set the ignore message the full line of the ignore directive if we've matched it successfully, though. (`ln`, here).

Reply:
Got it, it is my bad and not well considered.
By the way, thanks for squashing.

Reply:
Hi @Mark-Simulacrum 

The return value of `parse_name_value_directive` is already `Option<String>`, so this will not force the user to always has a reason to ignore a test case.  Such that the ignore attribute without reason presents, the test case still can be ignored, and the ignore_message will get `None` after parsing.

Here is `parse_name_value_directive`.
https://github.com/rust-lang/rust/blob/86b1581ea0e0e47196e1ea70b2d980b3eee72623/src/tools/compiletest/src/header.rs#L715-L724

I am willing to take your advice if there is something wrong, sorry about taking many rounds because this is my first PR here.

Thanks for your time in reviewing my PR :pray::heart:, I am willing to open PRs about ignoring issues of the test cases in Rust in the future, and participate in more development in Rust.